tests/components/pvpc_hourly_pricing/test_sensor.py

```diff
@@ -85,3 +109,65 @@ def mock_now():
             assert pvpc_aioclient_mock.call_count == 33
             assert len(caplog.messages) == 1
             assert caplog.records[0].levelno == logging.WARNING
+
+
+async def test_multi_sensor_migration(
+    hass, caplog, legacy_patchable_time, pvpc_aioclient_mock: AiohttpClientMocker
+):
+    """Test tariff migration when there are >1 old sensors."""
+    entity_reg = mock_registry(hass)
+    hass.config.time_zone = dt_util.get_time_zone("Europe/Madrid")
+    uid_1 = "discrimination"
+    uid_2 = "normal"
+    old_conf_1 = {CONF_NAME: "test_pvpc_1", ATTR_TARIFF: uid_1}
+    old_conf_2 = {CONF_NAME: "test_pvpc_2", ATTR_TARIFF: uid_2}
+
+    config_entry_1 = MockConfigEntry(domain=DOMAIN, data=old_conf_1, unique_id=uid_1)
+    config_entry_1.add_to_hass(hass)
+    entity1 = entity_reg.async_get_or_create(
+        domain="sensor",
+        platform=DOMAIN,
+        unique_id=uid_1,
+        config_entry=config_entry_1,
+        suggested_object_id="test_pvpc_1",
+    )
+
+    config_entry_2 = MockConfigEntry(domain=DOMAIN, data=old_conf_2, unique_id=uid_2)
+    config_entry_2.add_to_hass(hass)
+    entity2 = entity_reg.async_get_or_create(
+        domain="sensor",
+        platform=DOMAIN,
+        unique_id=uid_2,
+        config_entry=config_entry_2,
+        suggested_object_id="test_pvpc_2",
+    )
+
+    assert len(hass.config_entries.async_entries(DOMAIN)) == 2
+    assert len(entity_reg.entities) == 2
+
+    mock_data = {"return_time": datetime(2019, 10, 27, 20, tzinfo=date_util.UTC)}
+
+    def mock_now():
+        return mock_data["return_time"]
+
+    caplog.clear()
+    with caplog.at_level(logging.WARNING):
+        with patch("homeassistant.util.dt.utcnow", new=mock_now):
+            await hass.config_entries.async_setup(config_entry_1.entry_id)
+            assert len(caplog.messages) == 2
+            assert_setup_component(1, DOMAIN)
+
+            # check migration with removal of extra sensors
+            assert len(entity_reg.entities) == 1
+            assert entity1.entity_id in entity_reg.entities.keys()
```

Comment:
The keys is the default used for dict membership test.
```suggestion
            assert entity1.entity_id in entity_reg.entities
```